lib/edition_tagger.rb

```diff
@@ -7,7 +7,10 @@ def initialize(edition_tag_associations, logger)
 
   def run
     @edition_tag_associations.each do |association|
-      add_mainstream_browse_tag(association[:slug], association[:tag])
+      unless ["TRUE", "FALSE", nil].include? association[:primary]
+        raise "Invalid value for `primary`: #{association[:primary]}"
```

Comment:
Why do you expect an invalid value here? Can you add a test to show this intent?